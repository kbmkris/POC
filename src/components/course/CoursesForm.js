import React from "react";
import styles from "../../css/index.css";
import { PropTypes } from "prop-types";


const CoursesForm = ({
  allCoursesData,
  btnClassName,
  panelClassName,
  handleOnClickToggle,
  handleOnClickEnroll
}) => (
    <div className="main-div">
      {allCoursesData.map((allCoursesData,idx) => (
        <React.Fragment key={idx}>
          <button
            className={btnClassName[idx]}
            name={`button${idx}`}
            onClick={(e) => handleOnClickToggle(e,idx)}>
          {allCoursesData.courseName}
          </button>
          <div className={panelClassName[idx]} name={`div{idx}`}>
          <input
            type="button"
            className="button enroll-button"
            onClick={(e) => handleOnClickEnroll(e,allCoursesData.courseId, allCoursesData.courseName)}
            value="Enroll" />
            <ul className="list">
              {allCoursesData.title.map((title,idx) => (
                <React.Fragment key={idx}>
                  <li key={`l${idx}`}>
                    <span className="title-span">
                      {title.courseTitle + "   -   "}
                    </span>
                    <a target="_blank" href={title.link}>{title.link}</a>
                  </li>
                  <ul className="inner-list">
                    {title.topic.map((topic,tidx) => (
                      <li key={tidx}>{topic.topicName}</li>
                    ))}
                  </ul>
                </React.Fragment>
              ))}
            </ul>
          </div>
        </React.Fragment>
      ))}
    </div>
);


CoursesForm.propTypes = {
  allCoursesData : PropTypes.array.isRequired,
  btnClassName: PropTypes.array.isRequired,
  panelClassName: PropTypes.array.isRequired,
  handleOnClickToggle : PropTypes.func.isRequired,
  handleOnClickEnroll : PropTypes.func.isRequired
};

export default CoursesForm;
