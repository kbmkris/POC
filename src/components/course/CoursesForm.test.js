import React from "react";
import CoursesForm from "./CoursesForm";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";

describe("CoursesForm Test", () => {
  it("Snapshot - component should render", () => {
    const handleOnClickToggle = jest.fn();
    const handleOnClickEnroll = jest.fn();
    const component = renderer
      .create(
        <CoursesForm
          allCoursesData = {[{
            courseName: "",
            courseId: 0,
            title: [
              {
                courseTitleId: 0,
                courseTitle: "",
                link: "",
                courseId: 0,
                topic: [
                  {
                    topicId: 0,
                    topicName: ""
                  }
                ]
              }
            ]
          }]}
          btnClassName={["collapsible"]}
          panelClassName={["panelInactive"]}
          handleOnClickToggle = {handleOnClickToggle}
          handleOnClickEnroll = {handleOnClickEnroll}
        />,
      );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe ("Dom validation using enzyme", () => {
  function setup () {
    const props = {
      allCoursesData: [
        {
          courseName: "HTML",
          courseId: 1,
          title: [
            {
              courseTitleId: 1,
              courseTitle: "Fundamental",
              link: "https://stackoverflow.com/html",
              courseId: 1,
              topic: [
                {
                  topicId: 1,
                  topicName: "Basic tag"
                },
                {
                  topicId: 2,
                  topicName: "Form"
                }
              ]
            },
            {
              courseTitleId: 2,
              courseTitle: "Advanced Feature",
              link: "https://stackoverflow.com/advanced",
              courseId: 1,
              topic: [
                {
                  topicId: 3,
                  topicName: "Grid"
                }
              ]
            }
          ]
        }],
      btnClassName: ["collapsible"],
      panelClassName: ["panelInactive"],
      handleOnClickToggle : jest.fn(),
      handleOnClickEnroll : jest.fn()
    };
    return shallow(<CoursesForm {...props} />);
  }
  it("check for all elements inside CoursesForm", () => {
    const wrapper = setup();
    expect(wrapper.find("button").length).toBe(1); //one course
    expect(wrapper.find("input").length).toBe(1); //one enroll button
    expect(wrapper.find("a").length).toBe(2); //two anchor tags for link
    expect(wrapper.find("li").length).toBe(5); // no of lists = titles + topics
    //get initial class name for course & course details
    expect(wrapper.find("button").hasClass("collapsible")).toEqual(true);
    expect(wrapper.find("div").at(1).hasClass("panelInactive")).toEqual(true);
  });
});
