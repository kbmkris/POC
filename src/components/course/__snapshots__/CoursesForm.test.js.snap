// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursesForm Test Snapshot - component should render 1`] = `
<div
  className="main-div"
>
  <button
    className="collapsible"
    name="button0"
    onClick={[Function]}
  >
    
  </button>
  <div
    className="panelInactive"
    name="div{idx}"
  >
    <input
      className="button enroll-button"
      onClick={[Function]}
      type="button"
      value="Enroll"
    />
    <ul
      className="list"
    >
      <li>
        <span
          className="title-span"
        >
             -   
        </span>
        <a
          href=""
          target="_blank"
        >
          
        </a>
      </li>
      <ul
        className="inner-list"
      >
        <li>
          
        </li>
      </ul>
    </ul>
  </div>
</div>
`;
